server/util.py

```diff
@@ -0,0 +1,257 @@
+"""
+Utility methods for analysis-runner server
+"""
+import os
+import json
+from shlex import quote
+
+from aiohttp import web, ClientSession
+from hailtop.config import get_deploy_config
+from google.cloud import secretmanager, pubsub_v1
+
+from cpg_utils.cloud import email_from_id_token
+
+GITHUB_ORG = 'populationgenomics'
+METADATA_PREFIX = '/tmp/metadata'
+PUBSUB_TOPIC = 'projects/analysis-runner/topics/submissions'
+DRIVER_IMAGE = os.getenv('DRIVER_IMAGE')
+assert DRIVER_IMAGE
+
+COMBINE_METADATA = """
+import json
+import sys
+
+def load(filename):
+    text = open(filename).read().strip()
+    val = json.loads(text) if len(text) else []
+    return val if type(val) is list else [val]
+
+print(json.dumps(load(sys.argv[1]) + load(sys.argv[2])))
+"""
+
+secret_manager = secretmanager.SecretManagerServiceClient()
+publisher = pubsub_v1.PublisherClient()
+
+
+def _read_secret(name: str) -> str:
+    """Reads the latest version of the given secret from Google's Secret Manager."""
+    secret_name = f'projects/analysis-runner/secrets/{name}/versions/latest'
+    response = secret_manager.access_secret_version(request={'name': secret_name})
+    return response.payload.data.decode('UTF-8')
+
+
+server_config = json.loads(_read_secret('server-config'))
```

Comment:
Let's fetch this every time, otherwise stale data will become hard to debug. This won't take long at all.